soc/xtensa/intel_adsp/common/include/cavs-link.ld

```diff
@@ -410,12 +410,13 @@ SECTIONS {
   . = SEGSTART_UNCACHED;
   _heap_sentry = .;
 
-  /* dma buffers */
-  .lpbuf (NOLOAD): {
+  /* dma buffers, require 128 byte alignment */
+  .hpbuf (NOLOAD): {
+    . = ALIGN(128);
```

Comment:
Note that the modern SOF model is to allocate DMA memory from a unified heap, and the intent is that this will be extended via the multiheap model.  Putting buffers into static linker-managed sections is probably not what we want long term.  If you need to define a static buffer, just an "aligned(128)" attribute (or a soc-level macro to abstract that) is probably preferable.

I can't remember why the lpbuf section was left here.  IIRC there were some references to (but not uses of?) the section in the SOF source still, or something like that.

Reply:
It was used in the s1000 I believe with the designware dma, could be wrong here though. It seems nice to have a dedicated linker section to dma buffers to me and it was already there when I got here, I just tweaked it to move where it was placed.

Certainly another, more integrated, test case might do a heap allocation, send over a loop with hda, and validate it end to end with the DMA API involved, we'll get there.